lib/src/container/mod.rs

```diff
@@ -112,6 +154,69 @@ impl TryFrom<&str> for ImageReference {
     }
 }
 
+impl TryFrom<&str> for SignatureSource {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> Result<Self> {
+        match value {
+            "ostree-image-signed" => Ok(Self::ContainerPolicy),
+            "ostree-unverified-image" => Ok(Self::ContainerPolicyAllowInsecure),
+            o => match o.strip_prefix("ostree-remote-image:") {
+                Some(rest) => Ok(Self::OstreeRemote(rest.to_string())),
+                _ => Err(anyhow!("Invalid signature source: {}", o)),
+            },
+        }
+    }
+}
+
+impl TryFrom<&str> for OstreeImageReference {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> Result<Self> {
+        let mut parts = value.splitn(2, ':');
+        // Safety: Split always returns at least one value.
+        let first = parts.next().unwrap();
+        let second = parts
+            .next()
+            .ok_or_else(|| anyhow!("Missing ':' in {}", value))?;
+        let (sigverify, rest) = match first {
+            "ostree-image-signed" => (SignatureSource::ContainerPolicy, Cow::Borrowed(second)),
+            "ostree-unverified-image" => (
+                SignatureSource::ContainerPolicyAllowInsecure,
+                Cow::Borrowed(second),
+            ),
+            "ostree-remote-registry" => {
+                let mut subparts = second.splitn(2, ':');
+                // Safety: Split always returns at least one value.
+                let remote = subparts.next().unwrap();
+                let rest = subparts
+                    .next()
+                    .ok_or_else(|| anyhow!("Missing second ':' in {}", value))?;
+                (
+                    SignatureSource::OstreeRemote(remote.to_string()),
+                    Cow::Owned(format!("registry:{}", rest)),
+                )
+            }
+            "ostree-remote-image" => {
+                let mut subparts = second.splitn(2, ':');
+                // Safety: Split always returns at least one value.
+                let remote = subparts.next().unwrap();
+                let second = Cow::Borrowed(
+                    subparts
+                        .next()
+                        .ok_or_else(|| anyhow!("Missing second ':' in {}", value))?,
+                );
+                (SignatureSource::OstreeRemote(remote.to_string()), second)
+            }
+            o => {
+                return Err(anyhow!("Invalid signature source: {}", o));
+            }
+        };
+        let imgref = rest.deref().try_into()?;
```

Comment:
I think we should add some checks to guarantee that this starts with exactly one of `registry:` or `docker://`. Or maybe strip the scheme completely in-memory and add a formatter to print it in a uniform way.

Reply:
This and your above point both require us to parse the image reference.  Which is doable but a not-small scope increase, because there are a lot of "transports" supported by skopeo and containers/image, e.g. `containers-storage:` and `oci-archive:` are also both used.

In the end, a malformed image reference is going to result in an error (from skopeo) - we'd just be moving that error up earlier.  (Which is valuable, but IMO not critical)